api/src/flows.ts

```diff
@@ -413,7 +416,23 @@ class FlowManager {
 
 			return { successor: operation.resolve, status: 'resolve', data: result ?? null, options };
 		} catch (error: unknown) {
-			return { successor: operation.reject, status: 'reject', data: error ?? null, options };
+			let errorMessage: Record<string, any>;
+			if (error instanceof Error) {
+				errorMessage = { message: error.message };
+			} else {
+				// Is the error a JSON string? If so, parse it and use that as the error
+				errorMessage = isValidJSON(String(error))
+					: // Explicit specification of the error properties since those aren't enumerable
+					  JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
+					: JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```

Comment:
```suggestion
					? parseJSON(String(error))
					: // Explicit specification of the error properties since those aren't enumerable
					  JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```